packages/vite-runtime-workerd-plugin/src/index.ts

```diff
@@ -70,17 +94,109 @@ function getCreateRequestDispatcher(server: ViteDevServer) {
   return createRequestDispatcher;
 }
 
-function getMiniflareOptions() {
+async function getMiniflareOptions() {
+  const inspectorPort = await setupInspectorProxy();
+
   return {
     script,
     modules: true,
     unsafeEvalBinding: 'UNSAFE_EVAL',
     compatibilityDate: '2024-02-08',
-    inspectorPort: 9229,
+    inspectorPort,
     ...getOptionsFromWranglerToml(),
   };
 }
 
+// Dispatch any buffered requests and empty the buffer
+async function drainBuffer(server: ViteDevServer) {
+  const serverBaseAddress = getServerBaseAddress(server);
+  for (const req of buffer) {
+    await mf.dispatchFetch(`${serverBaseAddress}${req.url}`);
+  }
+  buffer = [];
+}
+
+// Initialize (or re-initialize) the inspector proxy server. Returns the port
+// of the proxy target which we feed to miniflare's `inspectorPort`
+async function setupInspectorProxy() {
+  // Remove any existing listeners that call the current `proxy`
+  inspectorProxyServer.removeAllListeners('request');
+  inspectorProxyServer.removeAllListeners('upgrade');
+
+  // Dispose of the old proxy and recreate it, potentially with a new
+  // target port
+  if (inspectorProxy) {
+    inspectorProxy.removeAllListeners();
+    inspectorProxy.close();
+  }
+
+  // Find an available inspector port. This port will be passed directly
+  // to miniflare. We look for an avaiable one since a zombie workerd process
+  // might be hanging on to 9229. If it is, we don't want miniflare to hang
```

Comment:
why do we get zombie workerd processes by the way? should something get fixed in Miniflare?